crates/nu-utils/src/utils.rs

```diff
@@ -9,19 +9,21 @@ pub fn enable_vt_processing() -> Result<()> {
 
         pub const ENABLE_PROCESSED_OUTPUT: u32 = 0x0001;
         pub const ENABLE_VIRTUAL_TERMINAL_PROCESSING: u32 = 0x0004;
-        // let mask = ENABLE_VIRTUAL_TERMINAL_PROCESSING;
+        let mask = ENABLE_VIRTUAL_TERMINAL_PROCESSING;
 
         let console_mode = ConsoleMode::from(Handle::current_out_handle()?);
         let old_mode = console_mode.mode()?;
 
         // researching odd ansi behavior in windows terminal repo revealed that
         // enable_processed_output and enable_virtual_terminal_processing should be used
-        // also, instead of checking old_mode & mask, just set the mode already
 
-        // if old_mode & mask == 0 {
-        console_mode
-            .set_mode(old_mode | ENABLE_PROCESSED_OUTPUT | ENABLE_VIRTUAL_TERMINAL_PROCESSING)?;
-        // }
+        if old_mode & mask == 0 {
+            if let Err(err) = console_mode
+                .set_mode(old_mode | ENABLE_PROCESSED_OUTPUT | ENABLE_VIRTUAL_TERMINAL_PROCESSING)
+            {
+                return Err(err);
+            }
```

Comment:
Isn't that effectively the same as `?` in this case?

Reply:
ya, I think so. I've changed it like 5 times and lost track of where I started. LOL.